apps/local_laplacian_fixed/local_laplacian_fixed_generator.cpp

```diff
@@ -0,0 +1,189 @@
+        }
+
+        // Make the laplacian pyramid of the output
+        Func outLPyramid[maxJ];
+        for (int j = 0; j < J; j++) {
+            // inGPyramid is a 10-bit value stored in an int16_t, so this shouldn't overflow if levels <= 32
+            Expr level = inGPyramid[j](x, y) * (levels-1);
+
+            // Split it into integer and fractional parts
+            Expr li = clamp(level / 1024, 0, levels-2);
+            Expr lf = level % 1024;
+
+            // Turn lf into an 8-bit interpolant
+            lf = cast<uint8_t>(lf / 4);
+
+            // Linearly interpolate between the nearest processed pyramid levels
+            outLPyramid[j](x, y) = lerp(lPyramid[j](x, y, li), lPyramid[j](x, y, li+1), lf);
+        }
+
+        // Make the Gaussian pyramid of the output
+        Func outGPyramid[maxJ];
+        outGPyramid[J-1](x, y) = outLPyramid[J-1](x, y);
+        for (int j = J-2; j >= 0; j--) {
+            outGPyramid[j](x, y) = upsample(outGPyramid[j+1])(x, y) + outLPyramid[j](x, y);
+        }
+
+        Func color;
+        color(x, y, c) = fixed_8(x, y, c) + (outGPyramid[0](x, y) - gray(x, y)) / 4;
+
+
+        // Clamp and cast back to 16-bit
+        Func output;
+        output(x, y, c) = cast<uint16_t>(clamp(color(x, y, c), 0, 255));
+
+        /* THE SCHEDULE */
+        remap.compute_root();
+
+        if (get_target().has_gpu_feature()) {
+            // gpu schedule
+            output.compute_root().gpu_tile(x, y, 16, 8);
+            for (int j = 0; j < J; j++) {
+                int blockw = 16, blockh = 8;
+                if (j > 3) {
+                    blockw = 2;
+                    blockh = 2;
+                }
+                if (j > 0) {
+                    inGPyramid[j].compute_root().gpu_tile(x, y, blockw, blockh);
+                    gPyramid[j].compute_root().reorder(k, x, y).gpu_tile(x, y, blockw, blockh);
+                }
+                outGPyramid[j].compute_root().gpu_tile(x, y, blockw, blockh);
+            }
+        } else if (get_target().features_any_of({Target::HVX_64, Target::HVX_128})) {
+            // cpu schedule
+            Var yi;
+            int vec_lanes_16 = get_target().natural_vector_size(UInt(16));
+            int vec_lanes_8  = get_target().natural_vector_size(UInt(8));
+            output.parallel(y, 32).vectorize(x, vec_lanes_8);
+            gray.compute_root().parallel(y, 32).vectorize(x, vec_lanes_16);
+            for (int j = 0; j < 4; j++) {
+                if (j > 0) {
+                    inGPyramid[j]
+                        .compute_root().parallel(y, 32).vectorize(x, vec_lanes_16);
+                    gPyramid[j]
+                        .compute_root().reorder_storage(x, k, y)
+                        .reorder(k, y).parallel(y, 8).vectorize(x, vec_lanes_16);
+                }
+                outGPyramid[j].compute_root().parallel(y, 32).vectorize(x, vec_lanes_16);
+            }
+            for (int j = 4; j < J; j++) {
```

Comment:
I agree with Dillon: I think this would be even more useful if it was rolled in to the existing LLF app as a scheduling option.